Completed React part code

Scoutress opened the pull request.

Scoutress committed:
All React part code

Scoutress committed:
Created RESTful API

Working 90%
Not working:
GET /businesses/:businessId/bookings/date/:date:

Scoutress committed:
Merge branch 'main' into CodeReviewNode

Scoutress committed:
Fixed bug with API. Other minor changes

Scoutress committed:
Changed CRLT to LF and fixed ESLint problems

Scoutress committed:
Added libs & env variables for auth

Scoutress committed:
Separated config files from server.js

Scoutress committed:
Created config to ensure all env variables are checked

Scoutress committed:
Created user model and index.js with all configs in one place

Scoutress committed:
Added UserModel and Login/Register functionality

Scoutress committed:
Added auth method, secured some API's, other changes

rokasandreikenas commented on package.json:
labai keistai atrodo kai sujungi node.js ir react projekto dependency į vieną failą, aš nematęs taip dar. Kodėl toks pasirinkimas? Taip pat sukėlęs esi serverį į client dalį, kai pagal idėją tūrėtų būti atskiri folderiai.   
`/client`
`/server`
arba
`/front-end`
`/back-end`

rokasandreikenas commented on src/hooks/UseCategories.js:
```diff
@@ -0,0 +1,19 @@
+
+  useEffect(() => {
+    const fetchCategories = async () => {
+      const response = await fetch("http://localhost:3000/categories");
+      const data = await response.json();
+      setCategories(data);
```
- Galima panaudoti `axios` biblioteką fetchui
- Nėra try catch, jeigu gausi errorą, console raudonuos
- Gali papildyti loading statu

rokasandreikenas commented on src/hooks/UseFilteredServices.js:
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable indent */
```
susitvarkyti es-lint ir prettier taisykles taip, kad nereikėtų

rokasandreikenas commented on src/server/config/ConfigEnvVariables.js:
```diff
@@ -0,0 +1,33 @@
+import "dotenv/config";
+
+const { SERVER_PORT, DB_CONNECTION, TOKEN_SECRET, TOKEN_EXPIRATION } =
+  process.env;
+
+const missingParams = [];
+
+if (!SERVER_PORT) {
+  missingParams.push("SERVER_PORT");
+}
+
+if (!DB_CONNECTION) {
+  missingParams.push("DB_CONNECTION");
+}
+
+if (!TOKEN_SECRET) {
+  missingParams.push("TOKEN_SECRET");
+}
+
+if (!TOKEN_EXPIRATION) {
+  missingParams.push("TOKEN_EXPIRATION");
+}
```
gal kokį ciklą išeitų parašyt, kad tų ifų nereikėtų kas kartą pridėti?

rokasandreikenas commented on src/server/middleware/ErrorHandler.js:
```diff
@@ -0,0 +1,9 @@
+// eslint-disable-next-line no-unused-vars
```
gal eitų pridėti `_` prie parametro ir šito atsikratyti

rokasandreikenas commented on src/server/models/CategoryModel.js:
```diff
@@ -0,0 +1,41 @@
+
+categorySchema.set("toJSON", {
+  transform: (doc, ret) => {
+    ret.id = ret._id;
+    delete ret._id;
+    delete ret.__v;
+    return ret;
```
interesting... kodėl trinamas _id? tas `__v` yra mongoDb naudojamas versijai, jį galima ir kitais būdais nuimti, per schemą `UserSchema = new mongoose.Schema({
    nickname: String,
    reg_time: {type: Date, default: Date.now}
}, {
    versionKey: false // You should be aware of the outcome after set to false
});`

rokasandreikenas commented on src/server/users/UserModel.js:
```diff
@@ -0,0 +1,120 @@
+  },
+  {
+    timestamps: true,
+    versionKey: false,
```
čia jau gerai

rokasandreikenas commented on src/server/users/UserModel.js:
```diff
@@ -0,0 +1,120 @@
+const userSchema = new mongoose.Schema(
+  {
+    username: { type: String, required: true, unique: true },
+    firstname: { type: String, required: true },
+    lastname: { type: String, required: true },
```
firstName ir lastName

Scoutress committed:
Separated client and server files

Scoutress committed:
Minor changes

Scoutress committed:
Remake of env variables config file

Scoutress committed:
Added axios and some other changes

Added axios lib with functionality
Fixed some code and styling bugs
Added loading components functionality
